lib/active_merchant/billing/gateways/securion_pay.rb

```diff
@@ -0,0 +1,269 @@
+        'lost_or_stolen' => STANDARD_ERROR_CODE[:card_declined],
+        'suspected_fraud' => STANDARD_ERROR_CODE[:card_declined],
+        'expired_token' => STANDARD_ERROR_CODE[:card_declined]
+      }
+
+      def initialize(options={})
+        requires!(options, :secret_key)
+        @secret_key = options[:secret_key]
+
+        super
+      end
+
+      def purchase(money, payment, options={})
+        post = {}
+        post = create_post_for_auth_or_purchase(money, payment, options)
+        commit(:post, 'charges', post, options)
+      end
+
+      def authorize(money, payment, options={})
+        post = {}
+        post = create_post_for_auth_or_purchase(money, payment, options)
+        post[:captured] = "false"
+        commit(:post, 'charges', post, options)
+      end
+
+      def capture(money, authorization, options = {})
+        post = {}
+        add_amount(post, money, options)
+
+        commit(:post, "charges/#{CGI.escape(authorization)}/capture", post, options)
+      end
+
+      def refund(money, authorization, options = {})
+        post = {}
+        add_amount(post, money, options)
+        commit(:post, "charges/#{CGI.escape(authorization)}/refund", post, options)
+      end
+
+      def void(authorization, options = {})
+        commit(:post, "charges/#{CGI.escape(authorization)}/refund", {}, options)
+      end
+
+      def verify(credit_card, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      def store(payment, options = {})
+        post = {}
+        add_creditcard(post, payment, options)
+
+        post[:description] = options[:description] if options[:description]
+        post[:email] = options[:email] if options[:email]
+        if options[:customer]
+          update_customer(options[:customer], post, options)
+        else
+          MultiResponse.run(:first) do |r|
+            r.process { verify(payment, options) }
+            if r.params["id"].present?
+              r.process { commit(:post, "customers", post, options) }
+            end
+          end
+        end
+      end
+
+      def update_customer(customer_id, post, options = {})
+        commit(:post, "customers/#{CGI.escape(customer_id)}", post, options)
+      end
+
+      def update(customer_id, card_id, options = {})
+        commit(:post, "customers/#{CGI.escape(customer_id)}/cards/#{CGI.escape(card_id)}", options, options)
+      end
+
+      def unstore(customer_id, options = {})
+        if options[:card_id]
+          commit(:delete, "customers/#{CGI.escape(customer_id)}/cards/#{CGI.escape(options[:card_id])}", nil, options)
+        else
+          commit(:delete, "customers/#{CGI.escape(customer_id)}", nil, options)
+        end
+      end
+
+      def create_token(payment, options = {})
```

Comment:
Actually `create_token` API method is meant for use from JavaScript or mobile applications, so this seems to be out of the scope of ActiveMerchant. Removed it.